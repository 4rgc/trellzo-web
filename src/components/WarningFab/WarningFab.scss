.warningFab {
	position: absolute;
	right: 20px;
	bottom: 20px;
}

.message {
	display: -webkit-box;
	max-width: 200px;
	overflow: hidden;
	text-overflow: ellipsis;
	// TODO: replace these with 'line-clamp: 3' when it's released
	-webkit-line-clamp: 3;
	-webkit-box-orient: vertical;
	text-align: left;
	transition: max-width 0.4s, -webkit-line-clamp 0s;
}

.message:hover {
	max-width: 400px;
	-webkit-line-clamp: 10;
}

.message-container {
	background-color: lightgrey;
	color: #b00020;
	border-radius: 10px;
	padding: 5px;
	margin-bottom: 5px;
}

.icon {
	float: right;
}
