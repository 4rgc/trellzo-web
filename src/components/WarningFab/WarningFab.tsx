import { useState } from 'react';
import WarningIcon from '../../assets/warning.svg';
import './WarningFab.scss';

export interface IWarningFabProps {
	displayOnMessage?: boolean;
	message?: string;
	display?: boolean;
	onClick?: () => void;
}

const WarningFab: React.FC<IWarningFabProps> = ({
	displayOnMessage = false,
	message,
	display,
	onClick,
}) => {
	const [isOpen, setIsOpen] = useState<boolean>(false);

	const displayFab =
		(display === undefined && displayOnMessage && message) || display;

	return (
		<div
			className="warningFab"
			style={{
				display: displayFab ? 'block' : 'none',
			}}
		>
			{isOpen && (
				<div className="message-container">
					<div className="message">{message}</div>
				</div>
			)}
			<img
				className="icon"
				src={WarningIcon}
				alt="warning icon"
				onClick={onClick || (() => setIsOpen((v) => !v))}
			/>
		</div>
	);
};

export default WarningFab;
