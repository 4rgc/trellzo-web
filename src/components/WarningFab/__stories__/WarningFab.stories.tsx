import { StoryFn, Meta } from '@storybook/react';
import WarningFab from '../WarningFab';

export default {
	title: 'Components/WarningFab',
	argTypes: {
		display: {
			type: 'boolean',
		},
		displayOnMessage: {
			type: 'boolean',
		},
		message: {
			type: 'string',
		},
	},
} as Meta<typeof WarningFab>;

const Template: StoryFn<typeof WarningFab> = (args) => <WarningFab {...args} />;

export const Default = Template.bind({});

export const AlwaysDisplay = Template.bind({});
AlwaysDisplay.args = { display: true };

export const DisplayOnMessage = Template.bind({});
DisplayOnMessage.args = { displayOnMessage: true, message: 'hello' };

export const DisplayOnMessageNoMessage = Template.bind({});
DisplayOnMessageNoMessage.args = { displayOnMessage: true };

export const WithCustomAction = Template.bind({});
WithCustomAction.args = { display: true };
WithCustomAction.argTypes = { onClick: { action: 'clicked' } };
