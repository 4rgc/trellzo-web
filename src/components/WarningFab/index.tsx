import WarningFab from './WarningFab';
export default WarningFab;
