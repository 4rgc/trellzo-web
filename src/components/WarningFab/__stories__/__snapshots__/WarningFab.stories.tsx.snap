// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/WarningFab AlwaysDisplay smoke-test 1`] = `
<div class="warningFab"
     style="display: block;"
>
  <img class="icon"
       src="static/media/warning.4bd46afe88062a02011b3f369983264a.svg"
       alt="warning icon"
  >
</div>
`;

exports[`Components/WarningFab Default smoke-test 1`] = `
<div class="warningFab"
     style="display: none;"
>
  <img class="icon"
       src="static/media/warning.4bd46afe88062a02011b3f369983264a.svg"
       alt="warning icon"
  >
</div>
`;

exports[`Components/WarningFab DisplayOnMessage smoke-test 1`] = `
<div class="warningFab"
     style="display: block;"
>
  <img class="icon"
       src="static/media/warning.4bd46afe88062a02011b3f369983264a.svg"
       alt="warning icon"
  >
</div>
`;

exports[`Components/WarningFab DisplayOnMessageNoMessage smoke-test 1`] = `
<div class="warningFab"
     style="display: none;"
>
  <img class="icon"
       src="static/media/warning.4bd46afe88062a02011b3f369983264a.svg"
       alt="warning icon"
  >
</div>
`;

exports[`Components/WarningFab WithCustomAction smoke-test 1`] = `
<div class="warningFab"
     style="display: block;"
>
  <img class="icon"
       src="static/media/warning.4bd46afe88062a02011b3f369983264a.svg"
       alt="warning icon"
  >
</div>
`;
